ouroboros-consensus-cardano/tools/db-analyser/Analysis.hs

```diff
@@ -161,6 +183,56 @@ showEBBs AnalysisEnv { db, registry } = do
           _otherwise ->
             return () -- Skip regular blocks
 
+{-------------------------------------------------------------------------------
+  Analysis: store a ledger at specific slot
+-------------------------------------------------------------------------------}
+
+storeLedgerAtSlot ::
+     forall blk .
+     ( LgrDbSerialiseConstraints blk
+     , HasAnalysis blk
+     , LedgerSupportsProtocol blk
+     )
+  => SlotNo -> Analysis blk
+storeLedgerAtSlot slotNo (AnalysisEnv { db, registry, initLedger, cfg, hasFS }) = do
+    putStrLn $ "About to store snapshot of a ledger at " <>
+               show slotNo <> " " <>
+               "this will take a while..."
+    void $ processAll db registry GetBlock initLedger process
```

Comment:
Yes, it will continue on processing, I have it on my bucket list of 'nice to have things, if we have time we will implement them'. Right now you will have the ability to override it with a `--limit` argument which we are about to add to the tool. But that will be a workaround, not a real solution.